Lib/urllib/parse.py

```diff
@@ -497,9 +497,15 @@ def urlunsplit(components):
     This may result in a slightly different, but equivalent URL, if the URL that
     was parsed originally had unnecessary delimiters (for example, a ? with an
     empty query; the RFC states that these are equivalent)."""
-    scheme, netloc, url, query, fragment, _coerce_result = (
+    scheme, netloc, path, query, fragment, _coerce_result = (
                                           _coerce_args(*components))
-    if netloc or (scheme and scheme in uses_netloc and url[:2] != '//'):
+    if scheme in uses_netloc and path.startswith('//'):
```

Comment:
Reorder these test. `path.startswith('//')` is the fast operation, do it first. `scheme in uses_netloc` is potentially slow as uses_netloc is a list in suboptimal order rather than a set.